service/controller/v22/resource/encryptionkey/delete.go

```diff
@@ -19,5 +20,20 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
+	encryptionKey, err := r.encrypter.EncryptionKey(ctx, customObject)
+	if r.encrypter.IsKeyNotFound(err) {
+		// We can get here during deletion, if the key is already deleted we can
+		// safely exit.
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	cc.Status.Cluster.EncryptionKey = encryptionKey
```

Comment:
Setting the status for `EnsureDeleted`. Note that the error handling above is different. It might happen the key is already gone at some point during deletion.

Reply:
This smells. I don't think we need the encryption key during the deletions. If we do we shouldn't make time dependencies, but somehow communicate when it is safe to delete the key.

Reply:
As it is in `master` right now, the encryption key gets fetched in `GetDesiredState` in the `s3object` resource. This also includes delete objects. I just replicated what was there. Do you want to drop it for delete events? I cannot think of any problem with that.

Reply:
> the encryption key gets fetched in GetDesiredState

This is incorrect IMO.

Reply:
Yes I think we should drop it.